org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/ReplaceFileContentChange.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2017, 2020 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.ui.refactoring2;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.ltk.core.refactoring.Change;
+import org.eclipse.ltk.core.refactoring.RefactoringStatus;
+import org.eclipse.ltk.core.refactoring.resource.ResourceChange;
+import org.eclipse.xtext.xbase.lib.Exceptions;
+import org.eclipse.xtext.xbase.lib.Pure;
+
+/**
+ * @author koehnlein - Initial contribution and API
+ * @since 2.13
+ */
+public class ReplaceFileContentChange extends ResourceChange {
+
+	private final IFile file;
+
+	private final byte[] newContents;
+
+	private String name;
+
+	public ReplaceFileContentChange(String name, IFile file, byte[] newContents) {
+		this(file, newContents);
+		this.name = name;
+	}
+
+	public ReplaceFileContentChange(final IFile file, final byte[] newContents) {
+		super();
+		this.file = file;
+		this.newContents = newContents;
+	}
+
+	@Override
+	protected IResource getModifiedResource() {
+		return file;
+	}
+
+	@Override
+	public RefactoringStatus isValid(IProgressMonitor pm) throws CoreException, OperationCanceledException {
+		final RefactoringStatus result = new RefactoringStatus();
+		if (!file.exists()) {
+			result.addFatalError("File " + file.getFullPath() + " does not exist");
+		}
+		return result;
+	}
+
+	@Override
+	public Change perform(final IProgressMonitor pm) throws CoreException {
+		final byte[] oldContents = this.getOldContents();
+		final ByteArrayInputStream newContentsIS = new ByteArrayInputStream(this.newContents);
+		return TryWithResource.<ReplaceFileContentChange> tryWith(newContentsIS, () -> {
+			try {
+				file.setContents(newContentsIS, true, true, pm);
+				return new ReplaceFileContentChange(file, oldContents);
+			} catch (Throwable e) {
+				throw Exceptions.sneakyThrow(e);
```

Comment:
No sneaky and a better catch would be nice

Reply:
Would you recommend throwing a RuntimeException here? Also in lines 91 and 95?

Reply:
It should be thrown as a CoreException as indicated by the method signature.